DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_changing_default_pvid.py

```diff
@@ -0,0 +1,106 @@
+import asyncio
+import pytest
+
+from dent_os_testbed.utils.test_utils.tgen_utils import (
+    tgen_utils_get_dent_devices_with_tgen,
+    tgen_utils_setup_streams, tgen_utils_start_traffic,
+    tgen_utils_stop_traffic, tgen_utils_get_traffic_stats,
+    tgen_utils_get_loss, tgen_utils_dev_groups_from_config,
+    tgen_utils_traffic_generator_connect)
+from dent_os_testbed.utils.test_utils.br_utils import (
+    get_traffic_port_vlan_mapping,
+    configure_bridge_setup, configure_vlan_setup)
+
+
+pytestmark = [pytest.mark.suite_functional_vlan,
+              pytest.mark.asyncio,
+              pytest.mark.usefixtures("cleanup_bridges", "cleanup_tgen")]
+
+
+async def test_vlan_changing_default_pvid(testbed):
+    """
+    Test Name: Changing default VLAN pvid
+    Test Suite: suite_functional_vlan
+    Test Overview: Test broadcast packet forwarding with non-default pvid
+    Test Procedure:
+    1. Configure bridge; enslave ports to bridge. Set bridge, ports to `up` state
+    2. Add ports to VLANs: all ports untagged, same vlans
+        Ports 1, 2, 3, 4 : vlan 4094, untagged, pvid
+    3. Map tx_ports and rx_ports
+    4. Setup broadcast packet streams: untagged, tagged with port VLAN, tagged with wrong VLAN
+    5. Send traffic to rx_ports.
+    6. Verify traffic loss on rx_ports per port_map configuration
+        - traffic loss on all ports for traffic with wrong VLAN;
+        - no traffic loss on all ports for traffic with port VLAN;
+        - no traffic loss on all ports for untagged traffic
+    """
+
+    # 1. Configure bridge; enslave ports to bridge. Set bridge, ports to `up` state
+    tgen_dev, dent_devices = await tgen_utils_get_dent_devices_with_tgen(testbed, [], 4)
+    if not tgen_dev or not dent_devices:
+        print("The testbed does not have enough dent with tgn connections")
+        return
+    device = dent_devices[0].host_name
+    tg_ports = tgen_dev.links_dict[device][0]
+    dut_ports = tgen_dev.links_dict[device][1]
+    packet_vids = ['X', 4094, 2]  # VLAN tag number contained in the transmitted packet
+    non_default_pvid = packet_vids[-1]
+
+    await configure_bridge_setup(device, dut_ports, default_pvid=non_default_pvid)
+
+    # 2. Set links to vlans per port_map configuration (all ports untagged, same vlan)
+    port_map = ({"port": 0, "settings": [{"vlan": 4094, "untagged": True, "pvid": True}]},
+                {"port": 1, "settings": [{"vlan": 4094, "untagged": True, "pvid": True}]},
+                {"port": 2, "settings": [{"vlan": 4094, "untagged": True, "pvid": True}]},
+                {"port": 3, "settings": [{"vlan": 4094, "untagged": True, "pvid": True}]})
+    await configure_vlan_setup(device, port_map, dut_ports)
+
+    # 3. Map tx_ports and rx_ports
```

Comment:
As I understand, real mapping happens on the line 99 in function get_traffic_port_vlan_mapping